infoschema/cache.go

```diff
@@ -51,18 +59,48 @@ func (h *InfoCache) GetLatest() InfoSchema {
 	infoschema_metrics.GetLatestCounter.Inc()
 	if len(h.cache) > 0 {
 		infoschema_metrics.HitLatestCounter.Inc()
-		return h.cache[0]
+		return h.cache[0].infoschema
 	}
 	return nil
 }
 
+func (h *InfoCache) getSchemaByTimestampNoLock(ts uint64) (InfoSchema, error) {
```

Comment:
Maybe change the return values' type to `(InfoSchema, bool)` is better. `errors.New` will generate a stack which is not so efficient and in semantic return a bool value is the same design with `map` if a item not exist